spec/jobs/suitable_for_early_allocation_email_job_spec.rb

```diff
@@ -0,0 +1,114 @@
+        create(:early_allocation, :ineligible, created_within_referral_window: false, nomis_offender_id: offender.offender_no)
+
+        expect_any_instance_of(EarlyAllocationMailer).not_to receive(:review_early_allocation)
+        described_class.perform_now
+      end
+    end
+
+    context 'when offender is not allocated to a POM' do
+      let(:release_date) { Time.zone.today + 17.months }
+
+      before do
+        Allocation.first.destroy
+      end
+
+      it 'does not send email' do
+        create(:early_allocation, created_at: Time.zone.today - 9.months, updated_at: Time.zone.today - 9.months,
+               created_within_referral_window: false, nomis_offender_id: offender.offender_no)
+
+        expect_any_instance_of(EarlyAllocationMailer).not_to receive(:review_early_allocation)
+        described_class.perform_now
+      end
+    end
+
+    context 'when offender not found in NOMIS' do
+      let(:release_date) { Time.zone.today + 17.months }
+
+      before do
+        allow(OffenderService).to receive(:get_offender).and_return(nil)
+      end
+
+      it 'does not send email' do
+        create(:early_allocation, created_at: Time.zone.today - 9.months, updated_at: Time.zone.today - 9.months,
+               created_within_referral_window: false, nomis_offender_id:  offender.offender_no)
+
+        expect_any_instance_of(EarlyAllocationMailer).not_to receive(:review_early_allocation)
+        described_class.perform_now
+      end
+    end
+
+    context 'when offender has more than 18 months of sentence remaining' do
+      let(:release_date) { Time.zone.today + 19.months }
+
+      it 'does not send email' do
+        create(:early_allocation, created_at: Time.zone.today - 9.months, updated_at: Time.zone.today - 9.months,
+               created_within_referral_window: false, nomis_offender_id:  offender.offender_no)
+
+        expect_any_instance_of(EarlyAllocationMailer).not_to receive(:review_early_allocation)
+        described_class.perform_now
+      end
+    end
+
+    context 'when offender has exactly 18 months of sentence remaining' do
+      let(:release_date) { Time.zone.today + 18.months }
+      let(:pom) { build(:pom) }
+
+      before do
+        create(:allocation, nomis_offender_id: offender.offender_no, primary_pom_nomis_id: pom.staff_id, primary_pom_name: pom.full_name)
+        allow(PrisonOffenderManagerService).to receive(:get_pom_at).and_return(pom)
+      end
+
+      it 'sends email' do
+        create(:early_allocation, created_at: Time.zone.today - 9.months, updated_at: Time.zone.today - 9.months,
+               created_within_referral_window: false, nomis_offender_id:  offender.offender_no)
+
+        expect_any_instance_of(EarlyAllocationMailer).to receive(:review_early_allocation).with(
+          email: pom.email_address,
+          prisoner_name: offender.full_name,
+          start_page_link: "/prisons/#{offender.prison_id}/prisoners/#{offender.offender_no}/early_allocations",
```

Comment:
It looks like the start page link only holds the URL path right now (e.g. `/prisons/AAA/prisoners/BBB/early_allocations`) – but is missing the hostname from the beginning (e.g. `https://example.com`) so will not be a clickable link for users.

It looks like we should be able to use `prison_prisoner_early_allocations_url` rather than `prison_prisoner_early_allocations_path` and it give us a full absolute URL.

Reply:
Done